### PR TITLE
[Merged by Bors] - feat(Data/Real/EReal): add theorems 

### PR DESCRIPTION
**Pull request by pitmonticone**
- [x] Add `add_sub_cancel_right`
- [x] Add `right_distrib_of_nneg`
- [x] Add `left_distrib_of_nneg`
- [x] Add `le_iff_le_forall_real_gt`

Co-authored-by: @D-Thomine 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
> Reduce lines to avoid lint style error

Please can you revert these commits? This is not inviting you to try and compress the existing code as much as possible; it's best to leave the style warning in place, and discuss splitting the file on Zulip.

(probably the resolution here is to ignore the warning for now)

**Comment by pitmonticone** [comment]
Thank you @eric-wieser for your help. 

I've done as you asked. [Here](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Porting.20TopologicalEntropy.20into.20Mathlib/near/447002335) is the related Zulip message.

File: Mathlib/Data/Real/EReal.lean [hunk 1]
```diff
@@ -1239,6 +1245,71 @@ instance : HasDistribNeg EReal where
...
+    · rw [mul_top_of_pos a_pos, mul_top_of_pos b_pos, mul_top_of_pos (add_pos a_pos b_pos),
+        top_add_top]
+
+theorem left_distrib_of_nneg {a b c : EReal} (ha : 0 ≤ a) (hb : 0 ≤ b) :
+    c * (a + b) = c * a + c * b := by
+  nth_rewrite 1 [EReal.mul_comm]; nth_rewrite 2 [EReal.mul_comm]; nth_rewrite 3 [EReal.mul_comm]
+  exact right_distrib_of_nneg ha hb
+
+theorem le_iff_le_forall_real_gt (x y : EReal) :
+    y ≤ x ↔ ∀ (z : ℝ), (x < z) → (y ≤ z) := by
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Real/EReal.lean:1289]: Unless there's precedent elsewhere, I think the simpler expression should be on the right hand side.

Also, please remove superfluous parentheses

Comment (pitmonticone) [reply Mathlib/Data/Real/EReal.lean:1289]: Yes, there are precedents: 
- https://github.com/leanprover-community/mathlib4/blob/58c10cb5f7fdbedeb99512536e1079bb59990dcb/Mathlib/Algebra/Ring/Defs.lean
- https://github.com/leanprover-community/mathlib4/blob/58c10cb5f7fdbedeb99512536e1079bb59990dcb/Mathlib/Data/ZMod/Defs.lean
- https://github.com/leanprover-community/mathlib4/blob/58c10cb5f7fdbedeb99512536e1079bb59990dcb/Mathlib/NumberTheory/LucasLehmer.lean
- https://github.com/leanprover-community/mathlib4/blob/58c10cb5f7fdbedeb99512536e1079bb59990dcb/Mathlib/RingTheory/OreLocalization/Ring.lean
- https://github.com/leanprover-community/mathlib4/blob/58c10cb5f7fdbedeb99512536e1079bb59990dcb/Mathlib/SetTheory/Game/Basic.lean

The parentheses in `c * (a + b) = c * a + c * b` don't seem to be superfluous.

Comment (pitmonticone) [reply Mathlib/Data/Real/EReal.lean:1289]: Ops, sorry. Your comments are perfectly fine, but they were displayed aside the wrong line on VS Code.

Comment (pitmonticone) [reply Mathlib/Data/Real/EReal.lean]: Done. Thanks.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Pull request closed without merging**